### DISCUSSION TITLE
While Djing (in middle of a song) auto dj fades back in then fades back to live dj

### DISCUSSION THREAD

**Discussion by Torchwoodhosting** [savonet/liquidsoap#2498] [User Support]
Useing LiquidSoap V1.3.7 While Dj'ing (in middle of a song) auto dj fades back in then fades back to live dj the log shows this:

[input(dot)harbor_6469:4] Buffer emptied, buffering needed

i'm using this 

live = audio_to_stereo(input.harbor(icy=true,"/",port=4896,auth=auth,buffer=29.0,max=30.0,on_connect=live_connected(),on_disconnect=live_disconnected))
ignore(output.dummy(live, fallible=true))

It helps but still happens while streaming Does any one have an ideas how to fix this issue..... 

Its not the net issue, or drops nor does sams show any errors like unable to send data fast enough.... I'm at wits end here how to solve this issue...... Any help is appricated!

**Comment by toots** [comment]
Hi,

This is usually linked to a drop in the DJ's connection. The most straight forward tool is to increase the input buffer and see if that helps keeping the data flowing while the connection temporarily drops.

**Reply by Torchwoodhosting** [reply]
Thanks for the reply I have updated the setting to buffer=120.0,max=150.0 it did help some..... I agree with you that most of the time drops like this are caused by the dj's internet or lan dropping out however when it happens on dj's all over the world (US and UK) and some of them like my self have fiberoptic to the house and very high up and down streams not mention I have seen it happen like clock work say like every 6 min, or ever 12 mins... most drops wont be consistent like that. Some one has mentioned to me that they cause could be server the liquid soap is in its a duel intel xeon L5630 16 core dedicated server with each cpu running at 2.1Mhz with 48GB ram.....A friend had said that liquid soap wants servers that cpus to run at 3.0 or higher to prevent this issue This what they said exactly:: "I think it is a CPU clock problem, the liquidsoap does not like lower than @3.00 Ghz clock speed." is that true? To me it seems like liquidsoap shouldn't have a clock problem at 2.1mhz maybe id it was .5Mhz or or 1mhz.....(very old pcs LOL) but I just cant seem to figure this out, after searching the web for weeks I cant find any one else having the exact issue some that are like it....Anything thing else you or any on in this fiscussion forms might can suggest would be greatly apricated.

**Reply by toots** [reply]
Yes, this does look like a clock problem. We have an operator for these cases, `buffer.adaptative`, but it hasn't been tested in a while. Maybe you want to give it a try? See: https://www.liquidsoap.info/doc-2.0.6/reference.html#buffer.adaptative

**Reply by Torchwoodhosting** [reply]
I noticed that was in the V2.0.6 cook book is that also available in V1.3.7? right now I'm using the work around of having it buffer 2 to 3 mins of the stream and it seems be working there's been no drops however I have noticed starting out with a 2 min buffer after about 2 1/2 + hours the buffer/lag time ends up dropping to around 33 seconds.... if this is option is available in 1.3.7 I try it even thought the work around had helped some with the issue....
